docs/specs/src/hwp_4_formal.md

```diff
@@ -0,0 +1,403 @@
+
+**Ongoing Operation via Calls to Zome Functions:** All changes following genesis and initialization occur by Zome call to a function contained in the DNA in the following form:
+
+$$ Z_c = \{Z_f, Params, CapTokenSecret \} $$
+
+Where $Z_f$ is the Zome function being called, $Params$ are the parameters passed to that Zome function and $CapTokenSecret$ references the capabilities token which explicitly grants the calling agent the permission to rights to call that function.
+
+Based on the interface connection and state when the Zome call is received we construct a context which provides additional necessary parameters to validate state transformation:
+
+$$ Context(Z_c) = \{Provenance, C_n \} $$
+
+$Provenance$ contains the public key of the caller along with their cryptographic signature of the call as proof that it originated from the agent controlling the associated private key.
+
+$C_n$ is the source Chain's latest entry at the time we begin processing the zome call. The validation functions will all be called "as at" the state $C_n$. Since multiple simultaneous zome calls might be made, tracking the "as at" enables detection of another call having successfully changed the state of the chain before this call completed its execution, therefore validation may need to be tried again on the new chain state. 
+
+#### Zome Calls & Changing Local State
+
+First, Holochain's "subconscious" security system confirms the $CapTokenSecret$ permits the agent identified by the $Provenance$ to call the targeted function. It returns a failure if not.  Otherwise it proceeds to further check if the function was explicitly permitted by the referenced capability token.
+
+*Note on Permissions: Capabilities tokens function similarly to API keys. Cap token grants are explicitly saved as private entries on the granting agent's source chain and contain a secret used to call them. Cap token claims are saved on the calling agent's chain so they can be used later to make calls that execute the capabilities that have been granted.*
+
+If the Zome call is one which alters local state (distinct from a call that just reads from the chain or DHT state), we must construct a bundle of state changes that will attempt to be appended to the source chain in an atomic/transactional commit:
+
+$$ \Delta C ( C _n, Z _c ) \to C _{n+1} $$
+
+where a Chain is comprised of paired entries, $e^x$, and actions, $h^x$: (actions were formerly called headers and we're still using this *h* in our formal symbols)
+
+$$
+ C _{n+1} - C _n = 
+\begin{Bmatrix}
+    h ^\prime, h ^{\prime\prime} \dots {h ^x}  \\ 
+    e ^\prime, e ^{\prime\prime} \dots {e ^x} 
+\end{Bmatrix}
+$$
+
+*If the validation rules pass* for these state changes **and the current top of chain is still in state $C_n$** then the transaction is committed to the persistent store. and the chain updated as follows:
+
+$$
+C _n = 
+\begin{Bmatrix}
+    h ^\text{DNA} \dots h ^n  \\ 
+    e ^\text{DNA} \dots e ^n
+\end{Bmatrix}
+$$
+
```

Comment:
It seems we might be conflating *records* with *transactions*. Previously, $C_n$ seemed to represent the $n$-th chain record. But now, each successive $C_n$ seems to represent the result of calling another zome function, which could add many records to the chain. Perhaps all that's needed is to define the entire genesis as a single state change resulting in the initial chain state $C_0$ which contains all three records. 

I'm assuming this is written with the eventual move to transaction-per-record instead of actions-per-record in mind -- which I like, but I want to make sure we're not mixing current with future.

I also have some specific notational suggestions that I'll include in a separate PR.